Data race on a channel

## Flaky Test

### Which jobs are failing

```
==================
WARNING: DATA RACE
Read at 0x00c000535cf0 by goroutine 7:
  runtime.chansend()
      /opt/hostedtoolcache/go/1.16.13/x64/src/runtime/chan.go:158 +0x0
  github.com/tikv/pd/client.(*client).WatchGlobalConfig.func1()
      /home/runner/work/pd/pd/client/client.go:1826 +0x304

Previous write at 0x00c000535cf0 by goroutine 91:
  runtime.closechan()
      /opt/hostedtoolcache/go/1.16.13/x64/src/runtime/chan.go:355 +0x0
  github.com/tikv/pd/tests/server/global_config_test.(*GlobalConfigTestSuite).TestClientWatchCloseReceiverExternally()
      /home/runner/work/pd/pd/tests/server/global_config/global_config_test.go:198 +0x1a4
  runtime.call16()
      /opt/hostedtoolcache/go/1.16.13/x64/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.16.13/x64/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:739 +0x11d

Goroutine 7 (running) created at:
  github.com/tikv/pd/client.(*client).WatchGlobalConfig()
      /home/runner/work/pd/pd/client/client.go:1805 +0x246
  github.com/tikv/pd/tests/server/global_config_test.(*GlobalConfigTestSuite).TestClientWatchCloseReceiverExternally()
      /home/runner/work/pd/pd/tests/server/global_config/global_config_test.go:196 +0x108
  runtime.call16()
      /opt/hostedtoolcache/go/1.16.13/x64/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.16.13/x64/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:739 +0x11d

Goroutine 91 (finished) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).run()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:696 +0x144
  github.com/pingcap/check.Run()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/run.go:150 +0x5a
  github.com/pingcap/check.RunAll()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/run.go:113 +0xff6
  github.com/pingcap/check.TestingT()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/run.go:99 +0x756
  github.com/tikv/pd/tests/server/global_config_test.Test()
      /home/runner/work/pd/pd/tests/server/global_config/global_config_test.go:36 +0x38
  testing.tRunner()
      /opt/hostedtoolcache/go/1.16.13/x64/src/testing/testing.go:1193 +0x202
==================
Error: 1/14 07:42:45.201 +00:00] [ERROR] [client.go:1808] ["[pd] panic in client `WatchGlobalConfig`"] [error="send on closed channel"]
FAIL	github.com/tikv/pd/tests/server/global_config	4.391s
```

### CI link

https://github.com/tikv/pd/runs/4813928097?check_suite_focus=true

### Reason for failure (if possible)

Data race on a channel.

--- a/tests/server/global_config/global_config_test.go
+++ b/tests/server/global_config/global_config_test.go
@@ -236,12 +236,6 @@ func (s *GlobalConfigTestSuite) TestClientWatch(c *C) {
 	}
 }
 
-func (s *GlobalConfigTestSuite) TestClientWatchCloseReceiverExternally(c *C) {
-	wc, err := s.watchGlobalConfig(s.server.Context())
-	c.Assert(err, IsNil)
-	close(wc)
-}
-
 func (s *GlobalConfigTestSuite) TestClientWatchTimeout(c *C) {
 	ctx, cancel := context.WithCancel(s.server.Context())
 	wc, _ := s.watchGlobalConfig(ctx)

